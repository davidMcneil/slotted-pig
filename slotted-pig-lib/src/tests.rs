use anyhow::Result;
use insta::assert_yaml_snapshot;
use test_case::test_case;

use crate::{categorizer::Categorizer, transaction::Transaction};

#[test_case("tests/categorizer_empty.yaml", "tests/transactions_empty.csv", "empty"; "empty")]
#[test_case("tests/categorizer_simple.yaml", "tests/transactions_simple.csv", "simple"; "simple")]
fn test_categorizer(categorizer: &str, transactions: &str, name: &str) -> Result<()> {
    let categorizer = Categorizer::from_yaml_file(categorizer)?;
    let transactions = Transaction::from_csv_file(transactions)?;
    let (categorized, _uncategorized) = categorizer.categorize(&transactions);
    assert_yaml_snapshot!(name, categorized);
    Ok(())
}
