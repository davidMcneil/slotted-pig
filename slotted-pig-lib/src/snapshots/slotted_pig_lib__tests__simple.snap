---
source: slotted-pig-lib/src/tests.rs
expression: categorizer.categorize(&transactions)
---
categorized:
  - category: expenses
    count: 2
    total: "-30"
    absolute_total: "30"
    children:
      subcategories:
        - category: store
          count: 2
          total: "-30"
          absolute_total: "30"
          children:
            transactions:
              - amount: "-10"
                account: credit card
                description: store1
                time: "2024-02-24T20:10:59Z"
              - amount: "-20"
                account: credit card
                description: store2
                time: "2024-02-23T20:10:59Z"
  - category: income
    count: 1
    total: "5"
    absolute_total: "5"
    children:
      subcategories:
        - category: paycheck
          count: 1
          total: "5"
          absolute_total: "5"
          children:
            transactions:
              - amount: "5"
                account: checking
                description: paycheck
                time: "2024-02-01T20:10:59Z"
